src/Http/Middleware/CsrfProtectionMiddleware.php

```diff
@@ -241,13 +241,17 @@ protected function _validateToken(ServerRequestInterface $request): void
         $body = $request->getParsedBody();
         if (is_array($body) || $body instanceof ArrayAccess) {
             $post = Hash::get($body, $this->_config['field']);
-            if ($this->_compareToken($post, $cookie)) {
+            if (Security::constantEquals($post, $cookie)) {
+                return;
+            } else if ($this->_compareToken($post, $cookie)) {
```

Comment:
no else required after a return;

Reply:
They could also just be combined into a single if statement.